### ISSUE TITLE
`Sequence` type is improperly defined

### ISSUE THREAD

**Issue by Stoops-ML** [poliastro/czml3#162]
The `Sequence` type ([link](https://github.com/poliastro/czml3/blob/598fb9dd42a36f5d92780f4c0f7bf244a4ee8e7c/src/czml3/types.py#L521)) is too broadly defined as it accepts as input `list[Any]`. It isn't defined in the [CZML wiki](https://github.com/AnalyticalGraphicsInc/czml-writer/wiki), and the only use case I can see for it is for defining a list of `TimeInterval`s (e.g. [here](https://github.com/poliastro/czml3/blob/598fb9dd42a36f5d92780f4c0f7bf244a4ee8e7c/src/czml3/examples/simple.py#L40)) or a list of `IntervalValue`s (e.g. [here](https://github.com/poliastro/czml3/blob/598fb9dd42a36f5d92780f4c0f7bf244a4ee8e7c/src/czml3/examples/simple.py#L177)).

It seems more appropriate to replace the `Sequence` class with either:
1. `SequenceTime` (or something similar) class that accepts `list[TimeInterval] | list[IntervalValue]`
2. `SequenceTimeInterval` class and `SequenceIntervalValue` class

My preference is for option 1.

As a side note I think many people will get this class confused with the `Sequence` class from `collections.abc`.

@astrojuanlu let me know your thoughts and I'll submit a PR (along with some other bug fixes I've found recently).

**Comment by Stoops-ML** [comment]
@astrojuanlu bump

**Comment by astrojuanlu** [comment]
I agree, `Sequence` is too broad. Weird that it's not defined in the CZML wiki, wondering where did I take it from then.

I would find it weird that `SequenceTime` accepts `list[InternalValue]` too, can't we make `Sequence` accept `list[TimeInterval] | list[IntervalValue]`, find some other generic enough name, or otherwise define two classes?

**Comment by Stoops-ML** [comment]
From the [TimeIntervalCollection](https://github.com/AnalyticalGraphicsInc/czml-writer/wiki/TimeIntervalCollection) wiki:
> A collection of time intervals, specified in ISO8601 interval format.

I think `TimeIntervalCollection` that accepts `list[TimeInterval] | list[IntervalValue]` would be appropriate. `IntervalValue` is simply a wrapper of `TimeInterval`, so the name `TimeIntervalCollection` covers all bases.

Note that I can't find anything related to czml3's `IntervalValue` class in the wiki.

**Comment by Stoops-ML** [comment]
I also want to add that changing the class name from `Sequence` is important as the name also exists in the very widely used `collections.abc` library.

**Comment by astrojuanlu** [comment]
Good points. Then let's go ahead with your option 👍🏼
